### ISSUE TITLE
Mono build is broken on OSX due to incorrect FrameworkPathOverride paths

### ISSUE THREAD

**Issue by baronfel** [dotnet/fsharp#6180]
When building via `msbuild` or `dotnet build` or `dotnet msbuild`  on an OSX system, any net4XX-targeting build fails because the FSharpBuild.Directory.Build.props hard-code mono paths that are standard on Linux, not on OSX.  the meaningful lines are 45-50:
```
 <!-- mono -->
  <PropertyGroup Condition="'$(OS)' == 'Unix'">
    <MonoPackaging Condition="$(TargetFramework.StartsWith('net4'))">true</MonoPackaging>
    <FrameworkPathOverride Condition="'$(TargetFramework)' == 'net45'">/usr/lib/mono/4.5-api</FrameworkPathOverride>
    <FrameworkPathOverride Condition="'$(TargetFramework)' == 'net46'">/usr/lib/mono/4.6-api</FrameworkPathOverride>
  </PropertyGroup>
```

#### Repro steps

Provide the steps required to reproduce the problem

1. Clone down latest master
2. run `dotnet build -f net462 src/fsharp/FSharp.Build/FSharp.Build.fsproj` from the root directory

#### Expected behavior

The `FSharp.Build` dll is build for the net462 TFM.

#### Actual behavior

The `FSharp.Build` dll fails to build because it first builds FSharp.Core, and attempts to locate explicit references via the hard-coded `/usr/lib/mono/4.5-api` path.  An example of the failing resolution is:
```
12:48:19.065  2:17>/Users/chethusk/oss/fsharp/artifacts/toolset/dotnet/sdk/2.1.500/Microsoft.Common.CurrentVersion.targets(2110,5): warning MSB3245: Could not resolve this reference. Could not locate the assembly "System". Check to make sure the assembly exists on disk. If this reference is required by your code, you may get compilation errors. [/Users/chethusk/oss/fsharp/src/fsharp/FSharp.Core/FSharp.Core.fsproj]
                             For SearchPath "{TargetFrameworkDirectory}". (TaskId:115)
                             Considered "/usr/lib/mono/4.5-api/System.winmd", but it didn't exist. (TaskId:115)
                             Considered "/usr/lib/mono/4.5-api/System.dll", but it didn't exist. (TaskId:115)
                             Considered "/usr/lib/mono/4.5-api/System.exe", but it didn't exist. (TaskId:115)
                             For SearchPath "{RawFileName}". (TaskId:115)
                             Considered treating "System" as a file name, but it didn't exist. (TaskId:115)
                     Primary reference "System.Data". (TaskId:115)

```

#### Known workarounds
None, because this property replacement overrides any `FrameworkPathOverride` that the user may set, eg via environmetn variable.

#### Related information 

Provide any related information 

* Operating system: OSX Mojave
* Branch: master
* .NET Runtime, CoreCLR or Mono Version: Mono 5.18 or 5.23, dotnet sdk 2.1.500 (as downloaded via the dotnet-install.sh script)
* Editing Tools (e.g. Visual Studio Version): VS Code w/ Ionide
* Indications of severity: high, this impacts the ability to merge and build on the `fsharp/fsharp` packaging repository.

**Comment by baronfel** [comment]
Note that this _should_ be able to be set on a per-OS basis by using MSBuild builtins for platform detection: https://github.com/Microsoft/msbuild/issues/2468#issuecomment-451194455

I've tried this locally myself and haven't been successful yet.  In addition, we may not want to set this value explicitly if it's possible to detect that the user has set an override already.  Finally, this matrix of reference directories should be expanded to the other -api folders:
<img width="96" alt="image" src="https://user-images.githubusercontent.com/573979/52181216-3ef02480-27b5-11e9-92d4-ac17f6471fdf.png">


I'd be glad to contribute all of this, just want to raise a flag.

**Comment by dsyme** [comment]
Fixed in #6181
